mypy/dmypy_server.py

```diff
@@ -135,10 +135,10 @@ def process_start_options(flags: List[str], allow_sources: bool) -> Options:
         ['-i'] + flags, require_targets=False, server_options=True
     )
     if options.report_dirs:
-        sys.exit("dmypy: start/restart cannot generate reports")
+        print("dmypy: ignoring report generation settings. Start/restart cannot generate reports.")
     if options.junit_xml:
-        sys.exit("dmypy: start/restart does not support --junit-xml; "
-                 "pass it to check/recheck instead")
+        print("dmypy: ignoring report generation settings. "
+              "Start/restart does not support --junit-xml. Pass it to check/recheck instead")
```

Comment:
Nit: Use capitalized 'Ignoring'.

I think it would be better to generate this warning and set `options.junit_xml` to `False`, even when using check/recheck.

Reply:
Updated. Thanks for the feedback, @JukkaL